### PR TITLE
feat: cosmos claim vertical stepper

### LINKED ISSUE #2329: Cosmos UI - Matching claim stepper to vertical deposit and withdraw implementation
### Overview

Gm!

Currently in the Cosmos UI, the deposit and withdraw steppers are vertical (shown below) and are awesome.
Screen Shot 2022-08-03 at 11 51 50 AM

We want to have the 'Claim' flow (shown below) mimic the same vertical stepper as the deposit and withdraw steppers instead of being horizontal.
Screen Shot 2022-08-03 at 11 51 58 AM

If mocks are needed for this, please let me or @reallybeard know.

### References and additional details

Placeholder

### Acceptance Criteria

- Cosmos claim flow is a vertical stepper matching the deposit and withdraw flows

### Need By Date

_No response_

### Screenshots/Mockups

_No response_

### Estimated effort

_No response_

### PR DESCRIPTION
**Pull request by stackedq**
## Description

Cosmos claim vertical stepper

## Notice

- [x] Have you followed the guidelines in our [Contributing]('https://github.com/shapeshift/web/CONTRIBUTING.md) guide?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/shapeshift/web/pulls) for the same update/change?

## Pull Request Type

- [ ] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [x] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

closes #2329

## Risk

Make sure claiming process is working as expected.

## Testing

open up Cosmos staking modal, click on the `claim` button, and you should see a vertical stepper instead of the old horizontal one.

## Screenshots (if applicable)

Screen Shot 2022-08-21 at 10 14 50
Screen Shot 2022-08-21 at 10 15 00

### CODE REVIEW

File: src/components/DeFi/components/Steps.tsx [hunk 1]
```diff
@@ -20,15 +20,19 @@ export type DefiStepProps = { [key in DefiStep]?: StepConfig }
 type StepsProps = {
   steps: DefiStepProps
   initialStep?: DefiStep
+  keepStatusInTheStepper?: boolean
```
Comment (0xApotheosis) [inline src/components/DeFi/components/Steps.tsx:23]:
```suggestion
  persistStepperStatus?: boolean
```

File: src/components/DeFi/components/Steps.tsx [hunk 2]
```diff
@@ -20,15 +20,19 @@ export type DefiStepProps = { [key in DefiStep]?: StepConfig }
...
   steps,
   children,
   initialStep = DefiStep.Info,
+  keepStatusInTheStepper,
   ...rest
 }) => {
-  const otherSteps = Object.keys(steps).filter(e => e !== DefiStep.Status)
+  const otherSteps = Object.keys(steps).filter(e =>
+    keepStatusInTheStepper ? true : e !== DefiStep.Status,
+  )
```
Comment (0xApotheosis) [inline src/components/DeFi/components/Steps.tsx:35]: We could make this a one-liner.

```suggestion
  const otherSteps = Object.keys(steps).filter(e => keepStatusInTheStepper ?? e !== DefiStep.Status)
```

File: src/features/defi/providers/cosmos/components/CosmosManager/Claim/components/Confirm.tsx [hunk 3]
```diff
@@ -0,0 +1,191 @@
...
+          </Row.Value>
+        </Row>
+        <Row>
+          <Row.Label>
+            <Text translation='common.estimatedGas' />
+          </Row.Label>
+          <Row.Value>
+            <SkeletonText
+              noOfLines={2}
+              isLoaded={!!bnOrZero(state.claim.estimatedGasCrypto).gt(0)}
```
Comment (0xApotheosis) [inline src/features/defi/providers/cosmos/components/CosmosManager/Claim/components/Confirm.tsx:156]: `gt()` already returns a boolean, no need to cast as truthy falsy.

```suggestion
              isLoaded={bnOrZero(state.claim.estimatedGasCrypto).gt(0)}
```

File: src/features/defi/providers/cosmos/components/CosmosManager/Claim/components/Status.tsx [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+  const opportunity = state?.opportunity
+  const { query, history } = useBrowserRouter<DefiQueryParams, DefiParams>()
+  const { chainId, assetReference } = query
+  const translate = useTranslate()
+  const assetNamespace = 'slip44'
+  const assetId = toAssetId({ chainId, assetNamespace, assetReference })
+  // Asset Info
+  const asset = useAppSelector(state => selectAssetById(state, assetId)) // TODO: diff denom for rewards
+  const txStatus = useMemo(
+    () => (!state ? TxStatus.PENDING : state.txid ? TxStatus.SUCCESS : TxStatus.FAILED),
```
Comment (0xApotheosis) [inline src/features/defi/providers/cosmos/components/CosmosManager/Claim/components/Status.tsx:50]: This is a tough pattern to mentally grok, I'd prefer an if else here.

**Review by 0xApotheosis — COMMENTED** [review]
Very tidy 👌 

I don't have any Cosmos to claim to test the stepper, but a few minor stylistic comments on the code.

**Review by gomesalexandre — APPROVED** [review]
LGTM, tested Cosmos and Osmosis claims and the new vertical stepper is used.

Note (not introduced by this PR), we should probably be better at handling loading states in DeFi, tried reloading the page at claim confirm state and I got this for Osmosis (with the flag off on reload):

image

and that state for Cosmos, staying in this state for a few seconds while Cosmos SDK data loads:

image

**Pull request merged**